assets/src/components/site-scan-results/site-scan-sources-list.js

```diff
@@ -39,49 +40,56 @@ export function SiteScanSourcesList( {
 					key={ slug }
 					className="site-scan-results__source"
 				>
-					{ name && (
-						<span className={ classnames( 'site-scan-results__source-name', {
-							'site-scan-results__source-name--inactive': [ 'inactive', 'uninstalled' ].includes( status ),
-						} ) }>
-							{ name }
-						</span>
-					) }
-					{ ! name && slug && (
-						<code className="site-scan-results__source-slug">
-							{ slug }
-						</code>
-					) }
-					{ status === 'active' ? (
-						<>
-							{ author && (
-								<span className="site-scan-results__source-author">
-									{ sprintf(
-										/* translators: %s is an author name. */
-										__( 'by %s', 'amp' ),
-										author,
-									) }
+					<details open={ false }>
+						<summary>
+							{ name && (
+								<span className={ classnames( 'site-scan-results__source-name', {
+									'site-scan-results__source-name--inactive': [ 'inactive', 'uninstalled' ].includes( status ),
+								} ) }>
+									{ name }
 								</span>
 							) }
-							{ version && (
-								<span className="site-scan-results__source-version">
-									{ sprintf(
-										/* translators: %s is a version number. */
-										__( 'Version %s', 'amp' ),
-										version,
+							{ ! name && slug && (
+								<code className="site-scan-results__source-slug">
+									{ slug }
+								</code>
+							) }
+							{ status === 'active' ? (
+								<>
+									{ author && (
+										<span className="site-scan-results__source-author">
+											{ sprintf(
+												/* translators: %s is an author name. */
+												__( 'by %s', 'amp' ),
+												author,
+											) }
+										</span>
 									) }
-								</span>
+									{ version && (
+										<span className="site-scan-results__source-version">
+											{ sprintf(
+												/* translators: %s is a version number. */
+												__( 'Version %s', 'amp' ),
+												version,
+											) }
+										</span>
+									) }
+								</>
+							) : (
+								<AMPNotice
+									className="site-scan-results__source-notice"
+									type={ NOTICE_TYPE_PLAIN }
+									size={ NOTICE_SIZE_SMALL }
+								>
+									{ status === 'inactive' ? inactiveSourceNotice : null }
+									{ status === 'uninstalled' ? uninstalledSourceNotice : null }
+								</AMPNotice>
 							) }
-						</>
-					) : (
-						<AMPNotice
-							className="site-scan-results__source-notice"
-							type={ NOTICE_TYPE_PLAIN }
-							size={ NOTICE_SIZE_SMALL }
-						>
-							{ status === 'inactive' ? inactiveSourceNotice : null }
-							{ status === 'uninstalled' ? uninstalledSourceNotice : null }
-						</AMPNotice>
-					) }
+						</summary>
+						<div className="site-scan-results__detail-body">
+							<SiteScanSourcesDetail slug={ slug } />
```

Comment:
Since `SiteScanSourcesDetail` component does some calculations on `scannableUrls`, it'd be best to not render it unless necessary (i.e. details are expanded). This could be achieved by removing the imperative `details`/`summary` elements pair and using a declarative approach, e.g.

```js
{ expandedPanel === slug && <SiteScanSourcesDetail slug={ slug } /> }
```

Reply:
I couldn't get you. The `details`/`summary` element is used to make it expandable. without using that it takes a lot of space in the display. And I don't think any theme/plugin will not have error detail.

Reply:
What I mean is that `details`/`summary` elements are handled client-side, entirely by a browser. Content of the `details` element should be in the markup from the very beginning, even if a particular panel is never opened by the user. That requires us to always render the `SiteScanSourcesDetail` component (where some calculations are done) while it may not be necessary at all.

If, instead, we would keep the expanded/collapsed state of the panel in a React state (with `useState()`) and render the `SiteScanSourcesDetail` if the panel is expanded, we would do the calculations only when needed.

This could be done by either controlling the content of the `details` element (similarly to this [example](https://codesandbox.io/s/2xyryvynqj)) or by using some other, stylized generic HTML elements instead of `details`/`summary`.

Anyway, on a second though, in this particular case it may be considered an over-engineering or premature optimization so feel free to disregard the original suggestion.

Reply:
@delawski I have made a change to render the content of details when a user opens the panel. Thought I have used a different approach than you suggested. Please have a look at it.